examples/src/bin/unbuffered-async-client.rs

```diff
@@ -0,0 +1,245 @@
+            conn.process_tls_records(&mut incoming_tls[..incoming_used])?;
+
+        match dbg!(state) {
+            ConnectionState::AppDataAvailable(mut state) => {
+                while let Some(res) = state.next_record() {
+                    let AppDataRecord {
+                        discard: new_discard,
+                        payload,
+                    } = res?;
+                    discard += new_discard;
+
+                    if payload.starts_with(b"HTTP") {
+                        let response = core::str::from_utf8(payload)?;
+                        let header = response
+                            .lines()
+                            .next()
+                            .unwrap_or(response);
+
+                        println!("{header}");
+                    } else {
+                        println!("(.. continued HTTP response ..)");
+                    }
+
+                    received_response = true;
+                }
+            }
+
+            ConnectionState::MustEncodeTlsData(mut state) => {
+                helpers::try_or_resize_and_retry(
+                    |out_buffer| state.encode(out_buffer),
+                    |e| {
+                        if let EncodeError::InsufficientSize(is) = &e {
+                            Ok(*is)
+                        } else {
+                            Err(e.into())
+                        }
+                    },
+                    outgoing_tls,
+                    &mut outgoing_used,
+                )?;
+            }
+
+            ConnectionState::MustTransmitTlsData(mut state) => {
+                if let Some(mut may_encrypt) = state.may_encrypt_app_data() {
+                    encrypt_http_request(
+                        &mut sent_request,
+                        &mut may_encrypt,
+                        outgoing_tls,
+                        &mut outgoing_used,
+                    );
+                }
+
+                send_tls(&mut sock, outgoing_tls, &mut outgoing_used).await?;
+                state.done();
+            }
+
+            ConnectionState::NeedsMoreTlsData { .. } => {
+                recv_tls(&mut sock, incoming_tls, &mut incoming_used).await?;
+            }
+
+            ConnectionState::TrafficTransit(mut may_encrypt) => {
+                if encrypt_http_request(
+                    &mut sent_request,
+                    &mut may_encrypt,
+                    outgoing_tls,
+                    &mut outgoing_used,
+                ) {
+                    send_tls(&mut sock, outgoing_tls, &mut outgoing_used).await?;
+                    recv_tls(&mut sock, incoming_tls, &mut incoming_used).await?;
+                } else if !received_response {
+                    // this happens in the TLS 1.3 case. the app-data was sent in the preceding
+                    // `MustTransmitTlsData` state. the server should have already a response which
+                    // we can read out from the socket
```

Comment:
I think this is the same typo I flagged in an earlier commit; "the server should have already" -> "the server should have already written"